cmake/autodetect_executors.cmake

```diff
@@ -3,45 +3,46 @@ set(GINKGO_HAS_MPI OFF)
 set(GINKGO_HAS_CUDA OFF)
 set(GINKGO_HAS_DPCPP OFF)
 set(GINKGO_HAS_HIP OFF)
-find_package(OpenMP 3.0)
-find_package(MPI)
+
 include(CheckLanguage)
-check_language(CUDA)
-try_compile(GKO_CAN_COMPILE_DPCPP ${PROJECT_BINARY_DIR}/dpcpp
-    SOURCES ${PROJECT_SOURCE_DIR}/dpcpp/test_dpcpp.dp.cpp
-    CXX_STANDARD 17)
 
-if(OpenMP_CXX_FOUND)
-    if(NOT DEFINED GINKGO_BUILD_OMP)
+if (NOT DEFINED GINKGO_BUILD_OMP)
+    find_package(OpenMP 3.0)
```

Comment:
I think CUDA also needs this `find_package`

Reply:
Is it done by check_language(CUDA)?